Add ability to compile freefem++ with Ipopt, freeyams, mmg3d_v4, mshmet, mumps_seq and superlu4
--- a/examples++-load/ff-Ipopt.cpp
+++ b/examples++-load/ff-Ipopt.cpp
@@ -23,7 +23,11 @@
  along with Freefem++; if not, write to the Free Software
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
-//ff-c++-LIBRARY-dep:  Ipopt mumps-seq blas  libseq  fc  
+//ff-c++-LIBRARY-dep:  Ipopt
+
+#ifndef HAVE_CSTDDEF
+#define HAVE_CSTDDEF
+#endif
 
 //using namespace std;
 #include "IpTNLP.hpp"
--- a/examples++-load/MUMPS_seq.cpp
+++ b/examples++-load/MUMPS_seq.cpp
@@ -38,7 +38,7 @@
 
 #include "ff++.hpp"
 
-#include "mpi.h"
+#include "mumps_seq/mpi.h"
 #include "dmumps_c.h"
 #include "zmumps_c.h"
 
--- a/examples++-load/mmg3d-v4.0.cpp
+++ b/examples++-load/mmg3d-v4.0.cpp
@@ -10,7 +10,7 @@
 
 //
 //   for automatic  compilation with ff-c++
-//ff-c++-LIBRARY-dep:   mmg3d-v4
+//ff-c++-LIBRARY-dep:   mmg3d_v4
 //ff-c++-cpp-dep: 
 //  
 
--- a/examples++-load/SuperLu.cpp
+++ b/examples++-load/SuperLu.cpp
@@ -1,7 +1,7 @@
 //   for automatic  compilation with ff-c++
-//ff-c++-LIBRARY-dep:   superlu blas 
+//ff-c++-LIBRARY-dep:   Superlu4 blas
 //ff-c++-cpp-dep: 
-//  for Super4.0 library   
+//  for Super4.0 library
 #include "ff++.hpp"
 /*
 #include  <iostream>
--- a/examples++-load/mshmet.cpp
+++ b/examples++-load/mshmet.cpp
@@ -8,7 +8,7 @@
 // E-MAIL   : jacques.morice@ann.jussieu.fr
 //
 //   for automatic  compilation with ff-c++
-//ff-c++-LIBRARY-dep:  mshmet libMesh
+//ff-c++-LIBRARY-dep:  mshmet
 //ff-c++-cpp-dep:
 //  
 
--- a/examples++-load/freeyams.cpp
+++ b/examples++-load/freeyams.cpp
@@ -8,7 +8,7 @@
 // E-MAIL   : jacques.morice@ann.jussieu.fr
 //
 //   for automatic  compilation with ff-c++
-//ff-c++-LIBRARY-dep:   freeyams libMesh
+//ff-c++-LIBRARY-dep:   freeyams
 //ff-c++-cpp-dep: 
 //  
 
